addons/event_product/models/event_registration.py

```diff
@@ -0,0 +1,22 @@
+from odoo import fields, models
+
+
+class EventRegistration(models.Model):
+    _inherit = ['event.registration']
+
+    sale_status = fields.Selection(string="Sale Status", selection=[
+            ('to_pay', 'Not Sold'),
+            ('sold', 'Sold'),
+            ('free', 'Free'),
+        ], compute="_compute_registration_status", compute_sudo=True, store=True, readonly=True, precompute=True)
+
+    def _has_order(self):
+        return False
+
+    # sale_status can be computed in pos_event and event_sale. We need to make it
+    # available in both modules, so its now in event module.
+    def _compute_registration_status(self):
+        if not self._has_order():
+            for reg in self:
+                reg.sale_status = 'free'
+                reg.state = 'open'
```

Comment:
Please don't.
Installing this module will reset the state of all existing registrations.
```suggestion
```

Reply:
Interesting you should say that, it's apparently what already happens...

Reply:
Or actually @davidmonnom you can just check if it's set in the first place as it's just supposed to be the default. No need to make it more complicated than needed...

So just
```py
for reg in self:
    if not reg.sale_status:
        reg.sale_status = 'free'
    if not reg.state:
        reg.state = 'open'
```